java/demo/src/main/java/com/google/demo/helper/TemplateHelper.java

```diff
@@ -0,0 +1,24 @@
+package com.google.demo.helper;
+
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.data.SoyTemplate;
+import com.google.template.soy.tofu.SoyTofu;
+import java.net.URL;
+
+public class TemplateHelper {
```

Comment:
Or you can even extend it for different kinds of templates, so you would have `InputTemplate` (extends `LibPhoneNumberTemplate`, contains a reference to `InputFormTemplates` and encapsulates all the logic for building that template inside). Same for `Result` (which can be even split to `SingleResult` and `FileResult` or something like that). I think that we would end up with much cleaner code.

Reply:
> How about making this a regular instantiable class (`LibPhoneNumberTemplate` for example?) that would get created with `templateName`, `namespace` and `soyTemplate` and expose `render()` and `getResource()` methods?

Agreed! `getResource()` I will not expose though, this is for internal logic!

> Or you can even extend it for different kinds of templates, so you would have `InputTemplate` (extends `LibPhoneNumberTemplate`, contains a reference to `InputFormTemplates` and encapsulates all the logic for building that template inside). Same for `Result` (which can be even split to `SingleResult` and `FileResult` or something like that). I think that we would end up with much cleaner code.

Fully agree with you that would make it much cleaner, but since they are autogenerated it would simply be overwritten next time there is a change in the template...

Reply:
But only `InputFormTemplates` and others in that package are auto-generated, no? You can simply keep a reference to that class and call its methods, not change anything in it. `InputTemplate` would only be a kind of wrapper around it to encapsulate the logic of working with that particular template. If it's too complicated, we don't need to do it, it's just a suggestion.

Reply:
you are right! much better to separate servlet logic and rendering logic